src/simple_cache_client.rs

```diff
@@ -154,16 +166,66 @@ impl SimpleCacheClient {
         default_ttl_seconds: NonZeroU64,
     ) -> Result<Self, MomentoError> {
         let data_endpoint = utils::get_claims(&auth_token).c;
+        let agent_value = format!("rust:{}", VERSION);
+        let momento_endpoints = MomentoEndpointsResolver::resolve(&auth_token, &None);
+        let control_client = SimpleCacheClient::build_control_client(
+            auth_token.clone(),
+            momento_endpoints.control_endpoint,
+            agent_value.clone(),
+        )
+        .await;
+        let data_client = SimpleCacheClient::build_data_client(
+            auth_token.clone(),
+            momento_endpoints.data_endpoint,
+            agent_value,
```

Comment:
why don't we need `agent_value.clone()` here but do for `build_control_client`?

Reply:
Rust has the concept called ownership. (more details [here](https://doc.rust-lang.org/book/ch04-01-what-is-ownership.html))
So the first time, I passed `auth_token` to `build_control_client`, the clone of `auth_token` gets passed, which I believe necessary so that I can pass the actual `auth_token` to `build_data_client`. 
Once `auth_token` gets passed a different scope, it's now "owned" by that scope so cannot pass it to anywhere else.
But cloning values takes up some memory space, so I think we want to avoid cloning as much as possible. But that's when `clippy` comes in handy. If it finds redundant use of clone, then it will be linted.